lib/runcible/base.rb

```diff
@@ -124,19 +124,26 @@ def format_payload_json(payload_hash)
 
     def process_response(response)
       begin
-        body = response.body == "null" ? nil : JSON.parse(response.body)
-        if body.respond_to? :with_indifferent_access
-          body = body.with_indifferent_access
-        elsif body.is_a? Array
-          body = body.map do |i|
-            i.respond_to?(:with_indifferent_access) ? i.with_indifferent_access : i
-          end
-        end
+        body = JSON.parse(response.body)
+        body = case body
+               when Hash
+                 body.respond_to?(:with_indifferent_access) ? body.with_indifferent_access : body
+               when Array
+                 body.map do |val|
+                   val.respond_to?(:with_indifferent_access) ? val.with_indifferent_access : val
+                 end
+               when NilClass
+                 ''
```

Comment:
Why are we using `''` now instead of `nil`? Because RestClient::Response converts body to `''` if it's nil anyway (see https://git.io/vKM9e).